services/department.js

```diff
@@ -1,25 +1,70 @@
 const Department = require('../models/Department');
 const validate = require('../validations/generalValidation');
+const ApplicationError = require('../errors/applicationError');
 
-const getAll = async () => {
-    return Department.find().sort("name");
+const getAll = async (request) => {
+    let departments = [];
+    const name = request.query.name
+    if (name) {
+        departments = await Department.find({ name: name }).select('-employees');
+        if (!departments) {
+            throw new ApplicationError(`Department with the name ${name} not found.`, 404)
+        }
+    }
+    if (!name) {
+        departments = await Department.find().select('-employees').sort('name');
+        if (!departments) {
+            throw new ApplicationError('There are no existing departments.', 404)
+        }
+    }
+    const page = parseInt(request.query.page) || 1;
```

Comment:
https://stackoverflow.com/questions/13935733/mongoose-limit-offset-and-count-query